### DISCUSSION TITLE
终端 Ctrl+C 无法中断程序

### DISCUSSION THREAD

**Discussion by chaoyue1217** [opensumi/core#2872] [Q&A] [answered]
![image](https://github.com/opensumi/core/assets/18662948/a8e0012c-ea8f-4ab8-a9b0-92ce6e78ed09)
自带终端中，Ctrl+C 复制文字没有问题，但是 Ctrl+C 无法中断程序

**Comment by erha19** [comment]
@chaoyue1217 这里感觉像是环境判断存在问题，如果是 Windows 环境下复制的快捷键应该是 ctrl+shift+c, 你可以排查定位一下原因，相关代码见：
https://github.com/opensumi/core/blob/18818dce9402e250c1a98443081985a4eda4875d/packages/terminal-next/src/browser/contribution/terminal.keybinding.ts#L26

**Reply by chaoyue1217** [reply]
通过keybinding排查解决了ctrl+shift+c的问题，但是ctrl+c还是无法中断程序，外层覆盖过ctrl+c，但是在 execute 方法里并没有捕获到任何输出，还有可能是什么原因把这个操作给覆盖了吗？

**Reply by erha19** [reply]
@chaoyue1217  可以在按 ctrl+c 快捷键时断点这个位置看一下具体执行命令 https://github.com/opensumi/core/blob/670f4f48a72b2cd827cba6c7d9d87c3c158eed4b/packages/core-browser/src/keybinding/keybinding.ts#L891

**Reply by chaoyue1217** [reply]
找到了问题所在，如果红框中的两行注释掉就可以触发xterm的 ctrl+c 中断程序操作，这个算是个bug吧
![image](https://github.com/opensumi/core/assets/18662948/ac72752e-05b5-417d-bd2b-a75c2f89fc4e)

**Reply by erha19** [reply]
@chaoyue1217 你加一个日志断点，看一下执行到这里的 `command.id` 有哪些？

**Reply by chaoyue1217** [reply]
只有一个electron.copy

**Reply by erha19** [reply, marked as answer]
@chaoyue1217 那这里的问题就是，在终端下，命中了 `electron.copy` 注册的快捷键 `ctrl+c`, 你可以检查一下代码内是否有相应的快捷键注册逻辑，需要进行调整或追加 `when` 条件来避免这种情况

**Reply by chaoyue1217** [reply]
通过when解决，感谢
